### PR TITLE
Fix several outdated/broken links in documentation

### PR DESCRIPTION
**Pull request by jmarkos**
I used this script https://gist.github.com/jmarkos/d7f74f53d41613756ab3 to find the broken links and fixed them. It still finds some 404 links because https://docs.jboss.org/infinispan/8.0/apidocs/ is incomplete (e.g. client.hotrod package is missing). I also used <infinispan.version>8.0</infinispan.version> in documentation/pom.xml, not 7.1
Maybe it could be used as part of the release script, or added as a unit test to documentation in the spirit of https://issues.jboss.org/browse/ISPN-3713?

The 1st commit is unrelated and just removes unused files I noticed.

### CODE REVIEW

**Comment by danberindei** [comment]
Nice, could you also add the script to the `bin` directory?

**Comment by danberindei** [comment]
Oops, I should have read the full description :)

I think it should be a part of the release process only, not a unit test that runs every time - sometimes we run tests without Internet access. But as a first step, it should be available for people to run manually.

**Comment by jmarkos** [comment]
Added the script to bin/ and updated the infinispan.version in documentation/pom.xml to 8.0

File: spring/spring/src/test/resources/org/infinispan/spring/provider/sample/CachingBookDaoContextTest.xml [hunk 1]
```diff
@@ -1,37 +0,0 @@
-<beans xmlns="http://www.springframework.org/schema/beans" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
```
Comment (wburns) [inline spring/spring/src/test/resources/org/infinispan/spring/provider/sample/CachingBookDaoContextTest.xml]: Hrmm these xml files, they are just no longer valid?

**Comment by wburns** [comment]
Great Job @jmarkos ! :+1:

File: spring/spring/src/test/resources/org/infinispan/spring/provider/sample/CachingBookDaoContextTest.xml (hunk 1 above)
Comment (jmarkos) [inline spring/spring/src/test/resources/org/infinispan/spring/provider/sample/CachingBookDaoContextTest.xml]: They were used by CachingBookDaoContextTest.java which was removed in 47372df . I also ran the spring tests without them and everything passed.

**Comment by galderz** [comment]
Integrated, thanks @jmarkos :)

**Pull request closed without merging**